### DISCUSSION TITLE
MultiArduino on same LAN network

### DISCUSSION THREAD

**Discussion by kadirozdinc** [JAndrassy/ArduinoOTA#61] [Q&A]
Hi juraj, I have a question. In case there are a large number of arduino mega (for example 50 mega) on the same network, is it possible to upload my sketch to each arduino mega ? Are there any limitations on this point ?

**Comment by JAndrassy** [comment]
every Arduino has its IP address. the upload is to IP address. doesn't matter how many others are in network.
but suppress the OTA port propagation with `#define NO_OTA_PORT` before the include. 50 network ports could be a problem in IDE.
and to upload run arduinoOTA tool from command line

**Reply by kadirozdinc** [reply]
Thanks for reply, so how i can run arduinoOTA tool from command line, what should I write in command line ? I have windows 10 OS and path is  C:\Program Files (x86)\Arduino\hardware\arduino\avr\platform.local.txt . if it doesn't inconvenience you, please describe how .

------ edit

Now I could upload my sketch into Mega over network using this code in command line

`C:\Program Files (x86)\Arduino\hardware\tools\avr\bin\arduinoOTA.exe -address 192.168.1.180 -port 65280 -username arduino -password password -sketch C:\Users\kozdinc\Desktop\OTEthernet\OTEthernet.ino.with_bootloader.arduino_mega.hex -upload /sketch`

but I can upload my code in command line ONLY ONCE Altough I can upload my sketch into mega over network in IDE for many times. I generate compiled binary named "OTEthernet.ino.with_bootloader.arduino_mega"

It seems the problem is about bootloader. Compiled binary has normal mega bootloader not optiboot one. What might be the problem ?

**Reply by JAndrassy** [reply]
you should upload .bin and without bootloader

**Reply by kadirozdinc** [reply]
Arduino cannot generate bin file for arduino mega in Arduino IDE. Is there any way of generating bin file ?

**Reply by JAndrassy** [reply]
the extras/avr/platform.local.txt has it
https://github.com/jandrassy/ArduinoOTA#atmega-support

**Reply by kadirozdinc** [reply]
I added platform.local.txt in extras to C:\Program Files (x86)\Arduino\hardware\arduino\avr location. But result is the same, it still generates hex file. (platform.local.txt file existed in the location but some line of codes were missing I replaced it with the one in extras)

edit -------

Finally I made it. I dig into codes and compared to the original platform.txt, I realized something, a line of code 

`recipe.output.tmp_file={build.project_name}.hexrecipe.output.save_file={build.project_name}.{build.variant}.hex`

I changed it like

`recipe.output.tmp_file={build.project_name}.binrecipe.output.save_file={build.project_name}.{build.variant}.bin`

I can generate bin file and in command line 

`cd C:\Program Files (x86)\Arduino\hardware\tools\avr\bin\arduinoOTA.exe -address 192.168.1.180 -port 65280 -username arduino -password password -sketch C:\Users\kozdinc\Desktop\OTEthernet\OTEthernet.ino.with_bootloader.arduino_mega.bin -upload /sketch -b`

I have arduino 1.8.13 version. Thanks for interests juraj.

**Reply by JAndrassy** [reply]
a joke: How to make the computer read a .doc file? Rename it to .mp3.

revert the change in platform.txt

hex **and** bin are generated if you have
```
## Create output (bin file)
recipe.objcopy.bin.pattern="{compiler.path}{compiler.elf2hex.cmd}" -O binary {compiler.elf2hex.extra_flags} "{build.path}/{build.project_name}.elf" "{build.path}/{build.project_name}.bin"
```
in platform.local.txt
upload the .bin
